plumbum/cli/application.py

```diff
@@ -618,6 +625,108 @@ def help(self):  # @ReservedAssignment
+            as the remainder of the line.
+            """
+            i = 0
+            while i < len(s) and s[i] == ' ': i += 1
+            return s[:i], s[i:]
+
+        def paragraphs(text):
+            """Yields each paragraph of text along with its initial and subsequent indentations to be used by
+            textwrap.TextWrapper.
+
+            Identifies list items from their first non-space character being one of bullets '-', '*', and '/'.
+            However, bullet '/' is invisible and is removed from the list item.
+
+            :param text: The text to separate into paragraphs
+            """
+
+            paragraph = None
+            initial_indent = ""
+            subsequent_indent = ""
+
+            def current():
+                """Yields the current result if present.
+                """
+                if paragraph:
+                    yield paragraph, initial_indent, subsequent_indent
+
+            for part in text.lstrip("\n").split("\n"):
+                indent, line = split_indentation(part)
+
+                if len(line) == 0:
+                    # Starting a new paragraph
+                    yield from current()
+                    yield "", "", ""
+
+                    paragraph = None
+                    initial_indent = ""
+                    subsequent_indent = ""
+                else:
+                    # Adding to current paragraph
+                    def is_list_item(line):
+                        """Returns true if the first element of 'line' is a bullet character.
+                        """
+                        bullets = [ '-', '*', '/' ]
+                        return line[0] in bullets
+
+                    def has_invisible_bullet(line):
+                        """Returns true if the first element of 'line' is the invisible bullet ('/').
+                        """
+                        return line[0] == '/'
+
+                    if is_list_item(line):
+                        # Done with current paragraph
+                        yield from current()
+
+                        if has_invisible_bullet(line):
+                            line = line[1:]
+
+                        paragraph = line
+                        initial_indent = indent
+
+                        # Calculate extra indentation for subsequent lines of this list item
+                        i = 1
+                        while i < len(line) and line[i] == ' ': i += 1
+                        subsequent_indent = indent + " " * i
+                    else:
+                        if not paragraph:
+                            # Start a new paragraph
+                            paragraph = line
+                            initial_indent = indent
+                            subsequent_indent = indent
+                        else:
+                            # Add to current paragraph
+                            paragraph = paragraph + ' ' + line
+
+            yield from current()
```

Comment:
Your production must not use Python 2 (must be nice...). This `yield from` should be very easy to convert to:

```python
for item in current()
    yield item
```

Since it is an iterator and not a generator.